Test/Unit/Model/CatalogIngestion/ProductEventManagerTest.php

```diff
@@ -0,0 +1,315 @@
+        $result = $this->productEventManager->deleteProductEvent(self::PRODUCT_ID);
+        $this->assertTrue($result);
+    }
+
+    /**
+     * @test
+     */
+    public function testPublishProductEventAsyncJob()
+    {
+        $amqpConfigExist = false;
+
+        if (class_exists('Magento\Framework\Amqp\Config')) {
+            $queueConfig = $this->deploymentConfig->getConfigData(AmqpConfig::QUEUE_CONFIG);
+            $amqpConfigExist = isset($queueConfig[AmqpConfig::AMQP_CONFIG]) ? true : false;
+        }
+
+        if ($this->moduleManger->isEnabled('Magento_AsynchronousOperations') && !$amqpConfigExist) {
+            $this->expectExceptionMessage('Unknown connection name amqp');
+        }
+
+        if (!$this->moduleManger->isEnabled('Magento_AsynchronousOperations') && !$amqpConfigExist) {
+            $this->expectExceptionMessage('Magento Asynchronous Operations is not supported on your magento version, please verify.');
+        }
+
+        $bulkId = $this->productEventManager->publishProductEventAsyncJob(
+            self::PRODUCT_ID,
+            ProductEventInterface::TYPE_UPDATE
+        );
+
+        if (!$this->moduleManger->isEnabled('Magento_AsynchronousOperations') && $amqpConfigExist) {
+            $this->assertNotEmpty($bulkId);
+        }
+    }
+
+    /**
+     * @test
+     */
+    public function testSendProductEvent()
+    {
+        $apiHelper = $this->createPartialMock(ApiHelper::class, ['sendRequest']);
+        $apiHelper->expects(self::once())->method('sendRequest')->willReturn(true);
+        TestHelper::setProperty($this->productEventManager, 'apiHelper', $apiHelper);
+        $product = $this->createProduct();
+
+        $this->productEventManager->publishProductEvent($product->getId(), ProductEventInterface::TYPE_UPDATE);
+        $productEvent = $this->productEventRepository->getByProductId($product->getId());
+        $this->productEventManager->sendProductEvent($productEvent);
+    }
+
+    /**
+     * @test
+     */
+    public function testRunInstantProductEvent_withoutAsync()
+    {
+        $store = $this->objectManager->get(StoreManagerInterface::class);
+        $websiteId = $store->getWebsite()->getId();
+        $configData = [
+            [
+                'path' => BoltConfig::XML_PATH_CATALOG_INGESTION_INSTANT_ENABLED,
+                'value' => 1,
+                'scope' => ScopeInterface::SCOPE_WEBSITES,
+                'scopeId' => $websiteId,
+            ],
+            [
+                'path' => BoltConfig::XML_PATH_CATALOG_INGESTION_INSTANT_ASYNC_ENABLED,
+                'value' => 0,
+                'scope' => ScopeInterface::SCOPE_WEBSITES,
+                'scopeId' => $websiteId,
+            ]
+        ];
+        TestUtils::setupBoltConfig($configData);
+
+        $apiHelper = $this->createPartialMock(ApiHelper::class, ['sendRequest']);
+        $apiHelper->expects(self::once())->method('sendRequest')->willReturn(true);
```

Comment:
i think is better to check all request data in ProductEventRequestBuilderTest::getRequest() class because product data is big, i think there is enough just know that the event was sent, we are just testing cron can get the list of product events and call requests of them.